src/detectors/B0ECAL/B0ECAL.cc

```diff
@@ -33,11 +33,11 @@ extern "C" {
           {"B0ECalRawHits"},
 #endif
           {
-            .eRes = {0.0 * sqrt(dd4hep::GeV), 0.02, 0.0 * dd4hep::GeV},
+            .eRes = {0.0326 * sqrt(dd4hep::GeV), 0.02, 0.0 * dd4hep::GeV},
             .tRes = 0.0 * dd4hep::ns,
             .threshold= 5.0 * dd4hep::MeV,
-            .capADC = 16384,
-            .dyRangeADC = 20 * dd4hep::GeV,
+            .capADC = 160000,
```

Comment:
We should also use a realistic ADC capacity. The 14 bit was ok, but this is not a realistic ADC capacity. Especially relevant since it is likely too large which will cause you to think you have better performance than warranted.

Reply:
We assume 6x6 mm2 SiPM with 15mircon pitch - resulting in (6/0.015)^2 = 160000 pixels per sensors. If 160000 photons in the sensor corresponds to 170GeV deposited energy (base on the simulation studies) we obtain:
- Stochastic noise of sqrt(170/160000) sqrt(GeV) = 0.0326 sqrt(GeV)

The threshold was also set with respect to number of photons in the sensor (at least 10 photons - at least 10 pixels with a signal).

I will change the ADC capacity to 14bits, 1 ADC count will be corresponding to 10MeV (instead of 1 MeV) and I will reduce the threshold to 1 (instead of 10)

Reply:
@michael-pitt Would you mind adding your nice explanations as a comment to the code? Thank you.

Reply:
Regarding the stochastic noise, it's not implemented yet, we might use https://github.com/eic/EICrecon/pull/1064, or ad-hoc smearing can be added in the digitzation.

Reply:
I added explanations in https://github.com/eic/EICrecon/pull/1623/commits/4a7bfabe069a6b369f9614317a8934eb339dd489:
https://github.com/eic/EICrecon/blob/4a7bfabe069a6b369f9614317a8934eb339dd489/src/detectors/B0ECAL/B0ECAL.cc#L36

https://github.com/eic/EICrecon/blob/4a7bfabe069a6b369f9614317a8934eb339dd489/src/detectors/B0ECAL/B0ECAL.cc#L59